src/engine.js

```diff
@@ -191,11 +192,13 @@ class Engine extends EventEmitter {
       return rule.evaluate(almanac).then((ruleResult) => {
         debug(`engine::run ruleResult:${ruleResult.result}`)
         if (ruleResult.result) {
-          this.emit('success', rule.event, almanac, ruleResult)
-          this.emit(rule.event.type, rule.event.params, almanac, ruleResult)
-          almanac.factValue('success-events', { event: rule.event })
+          almanac.factValue('success-events', { event: ruleResult.event })
```

Comment:
As `almanac.factValue` returns a promise, Shouldn't this go inside `Promise.all` array as well?

Reply:
yikes, that promise handling has been broken for a while. good catch!